natss/cmd/channel_dispatcher/main.go

```diff
@@ -58,14 +60,23 @@ func main() {
 	defer flush(logger)
 
 	// set up signals so we handle the first shutdown signal gracefully
-	stopCh := signals.SetupSignalHandler()
+	ctx := signals.NewContext()
 
 	cfg, err := clientcmd.BuildConfigFromFlags(*masterURL, *kubeconfig)
 	if err != nil {
 		logger.Fatalw("Error building kubeconfig", zap.Error(err))
 	}
-
-	natssDispatcher, err := dispatcher.NewDispatcher(util.GetDefaultNatssURL(), util.GetDefaultClusterID(), clientID, logger.Desugar())
+	dispatcherArgs := dispatcher.Args{
```

Comment:
Can you refactor the code here to use `sharedmain` code? Look at how it works in `KafkaChannel` https://github.com/knative/eventing-contrib/blob/master/kafka/channel/cmd/channel_dispatcher/main.go

Reply:
Concluded!
Could you take another look?